cognite/client/data_classes/sequences.py

```diff
@@ -145,3 +179,136 @@ def remove(self, value: List) -> SequenceUpdate:
 class SequenceList(CogniteResourceList):
     _RESOURCE = Sequence
     _UPDATE = SequenceUpdate
+
+
+class SequenceData:
+    """An object representing a list of rows from a sequence.
+
+    Args:
+        id (int): Id of the sequence the data belong to
+        external_id (str): External id of the sequence the data belong to
+        rows (List[dict]): combined row numbers and row data object from the API. If you pass this, row_numbers/values are ignored.
+        row_numbers (List[int]): The data row numbers.
+        values (List[List[ Union[int, str, float]]]): The data values, one row at a time.
+        columns: List[dict]: The column information, in the format returned by the API
+    """
+
+    def __init__(
+        self,
+        id: int = None,
+        external_id: str = None,
+        rows: List[dict] = None,
+        row_numbers: List[int] = None,
+        values: List[List[Union[int, str, float]]] = None,
+        columns: List[dict] = None,
+    ):
+        if rows:
+            row_numbers = [r["rowNumber"] for r in rows]
+            values = [r["values"] for r in rows]
+        self.id = id
+        self.external_id = external_id
+        self.row_numbers = row_numbers or []
+        self.values = values or []
+        self.columns = columns
+
+    def __str__(self):
+        return json.dumps(self.dump(), indent=4)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __len__(self) -> int:
+        return len(self.row_numbers)
+
+    def __eq__(self, other):
+        return (
+            type(self) == type(other)
+            and self.id == other.id
+            and self.external_id == other.external_id
+            and self.row_numbers == other.row_numbers
+            and self.values == other.values
+        )
+
+    def __getitem__(self, item) -> List[Union[int, str, float]]:
+        # slow, should be replaced by dict cache if it sees more than incidental use
+        return self.values[self.row_numbers.index(item)]
```

Comment:
May be slow, but I like this idea!

I think it should behave slightly differently though. If a slice is passed I think it should return a new SequenceData object containing only the requested values, instead of a list of lists. This should also be reflected in the type hint.

Reply:
It's hard to implement efficiently, and I'm not sure it will see use. Maybe something to revisit later.

Reply:
Why is that hard to implement efficiently? `__getitem__` should support accepting a slice, but if you want to revisit this later you should at the very least raise a TypeError if a slice is passed, now it will raise a more cryptic ValueError saying that the slice cannot be found in the list. I would not say that slicing a list is something that will not see use.